R/register-chunk-hook.R

```diff
@@ -0,0 +1,53 @@
+#' Use vcr in vignettes
+#'
+#' @description
+#' `setup_knitr()` registers a `use_cassette` hook so that you map
+#' a vcr cassette to a knitr chunk. Call it in your setup chunk:
+#'
+#' ````
+#' ```{r setup}
+#' #| include: false
+#' vcr::setup_knitr()
+#' ```
+#' ````
+#'
+#' Then in a chunk where you want to use a cassette, set the `cassette` chunk option
+#' to the name of the cassette:
+#'
+#' ````
+#' ```{r}
+#' #| cassette: name
+#' req <- httr2::request("http://r-project.org")
+#' resp <- httr2::req_perform(req)
+#' ```
+#' ````
+#'
+#' @param prefix An prefix for the cassette name to make cassettes unique across
+#'   vignettes. Defaults to the file name (sans extension) of the currently
+#'   executing vignette.
+#' @param dir Directory where to create the cassette file, relative to
+#'   the directory containing the vignette. Default: `"_vcr"``.
+#' @param ... Other arguments passed on to [insert_cassette()].
+#' @export
+setup_knitr <- function(prefix = NULL, dir = "_vcr", ...) {
+  check_string(prefix, allow_null = TRUE)
+  check_string(dir)
+
+  if (is.null(prefix)) {
```

Comment:
So you wouldn't prefer using `%||%`?

Reply:
I would usually, but here it makes for a long line that gets wrapped a bit awkwardly.